app.rb

```diff
@@ -0,0 +1,121 @@
+require 'sinatra/base'
+require 'sinatra/reloader'
+require_relative 'lib/database_connection'
+require_relative 'lib/peep_repository'
+require_relative 'lib/peep'
+require_relative 'lib/user_repository'
+require_relative 'lib/user'
+require 'date'
+require 'cgi'
+
+DatabaseConnection.connect
+
+class Application < Sinatra::Base
+  
+  configure :development do
+    register Sinatra::Reloader
+    # set :show_exceptions, false
+    set :raise_errors, true
+    enable :sessions
+  end
+  
+  get '/' do
+   return erb(:index)
+  end
+  
+  # ------
+  # /peeps
+  # ------
+
+  get '/peeps' do
+    @peeps = PeepRepository.new.all
+    return erb(:peeps)
+  end
+
+  get '/peeps/new' do
+    if session[:user_id] == nil
+      return erb(:'/sessions/login')   
+    else
+      return erb(:peeps_new)
+    end
+  end
+
+  get '/peeps/posted' do
+    return erb(:peeps_posted)
+  end
+
+  post '/peeps' do
+   
+    # currently missing input validation check(s)
+    peep = Peep.new         
+    peep.content = CGI::escapeHTML(params[:content])
+    peep.date_time = DateTime.now.strftime('%d/%m/%Y %H:%M:%S')
+    peep.user_f_name = session[:user_f_name]
+    peep.user_handle = session[:user_handle]
+
+    repo = PeepRepository.new
+    repo.create(peep)
+
+    return erb(:peeps_posted)
+  end
+
+  # ------
+  # /users
+  # ------
+
+  get '/users/signup' do
+    return erb(:signup)
+  end
+
+  get '/users/user-created' do
+    return erb(:user_created)
+  end
+
+  post '/users/signup' do
+    @user = User.new
+
+    @user.email = CGI::escapeHTML(params[:email])
+    @user.password = CGI::escapeHTML(params[:password])
+    @user.f_name = CGI::escapeHTML(params[:f_name].capitalize)
```

Comment:
You may also need to check for SQL injection, unsure if escapeHTML removes SQL commands.